src/rred_reports/reports/schools.py

```diff
@@ -70,51 +70,54 @@
 
 
 def school_filter(whole_dataframe: pd.DataFrame, school_id):
-    """Function to filter school"""
-    return whole_dataframe[whole_dataframe.school_id == school_id].copy()
+    """Function to filter out the Teacher Leaders and by school
 
+    Args: pd.DataFrame, school_id
 
-# specific filter for each table
-def filter_one(school_dataframe: pd.DataFrame):
-    """Filter for table one, for later use
-
-    Args: school_filter(pd.DataFrame)"""
-    return school_dataframe
-
-
-def filter_two(school_dataframe: pd.DataFrame):
-    """Filter for table two, for later use
-
-    Args: school_filter(pd.DataFrame)"""
-    return school_dataframe
+    Returns: pd.DataFrame filtered to show only RR teachers data per school
 
+    """
+    teacher_filtered = whole_dataframe[
+        (whole_dataframe.reg_rr_title == "RR Teacher + Support Role") | (whole_dataframe.reg_rr_title == "RR Teacher + Class Leader")
+    ]
+    return teacher_filtered[teacher_filtered.school_id == school_id].copy()
 
-def filter_three(school_dataframe: pd.DataFrame):
-    """Filter for table three, for later use
 
-    Args: school_filter(pd.DataFrame)"""
-    return school_dataframe
+# specific filter for each table
+def filter_for_one_two_five(school_dataframe: pd.DataFrame):
+    """Filter for table one, two and five: <entry_date> OR <exit_date> is after 31/7/21 and before 1/8/22
 
+    Args: school_filter(pd.DataFrame)
 
-def filter_four(school_dataframe: pd.DataFrame):
-    """Filter for table four, for later use
+    Returns: school_filter(pd.DataFrame) filtered by the reporting year
+    """
+    return school_dataframe[
+        ((school_dataframe["entry_date"] > "2021-07-31") & (school_dataframe["entry_date"] < "2022-08-01"))
+        | ((school_dataframe["exit_date"] > "2021-07-31") & (school_dataframe["exit_date"] < "2022-08-01"))
+    ]
 
-    Args: school_filter(pd.DataFrame)"""
-    return school_dataframe
 
+def filter_for_three_four(school_dataframe: pd.DataFrame):
+    """Filter for table three and four: ONLY on pupils whose <exit_date> is after 31/7/21 and before 1/8/22.
+    ONLY on data for pupils with 'Discontinued' OR 'Referred to school' in the <exit_outcome> column
 
-def filter_five(school_dataframe: pd.DataFrame):
-    """Filter for table five, for later use
+    Args: school_filter(pd.DataFrame)
 
-    Args: school_filter(pd.DataFrame)"""
-    return school_dataframe
+    Returns: school_filter(pd.DataFrame) filtered by exit_outcome and exit_date
+    """
+    return school_dataframe[(school_dataframe["exit_outcome"] == "Discontinued") | (school_dataframe["exit_outcome"] == "Referred to school")][

```

Comment:
I think we could use `.isin` here to make it a bit more clear, and same for other other times. 
```suggestion
    return school_dataframe[(school_dataframe["exit_outcome"].isin(["Discontinued", "Referred to school"]) & 

```